src/ray/core_worker/core_worker.cc

```diff
@@ -516,17 +537,17 @@ Status CoreWorker::Wait(const std::vector<ObjectID> &ids, int num_objects,
   // Wait from both store providers with timeout set to 0. This is to avoid the case
   // where we might use up the entire timeout on trying to get objects from one store
   // provider before even trying another (which might have all of the objects available).
-  if (plasma_object_ids.size() > 0) {
-    RAY_RETURN_NOT_OK(plasma_store_provider_->Wait(
-        plasma_object_ids, num_objects, /*timeout_ms=*/0, worker_context_, &ready));
+  if (memory_object_ids.size() > 0) {
+    RAY_RETURN_NOT_OK(memory_store_->Wait(memory_object_ids, num_objects,
```

Comment:
@ericl  this seems like a bug,  it's possible that `num_objects` is larger than `memory_object_ids.size()`. The issue applies for the following `Wait` calls below for memory store and plasma store. 

A possible fix is to specify `std::min(memory_object_ids.size(), num_objects)` instead of `num_objects` for the second param.